File: citizen app/backend/api/app/schemas/location.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import SavedLocationLabel


class SavedLocationCreateRequest(BaseModel):
    label: SavedLocationLabel
    custom_label: str | None = Field(default=None, max_length=64)
    address: str
    latitude: float = Field(ge=-90, le=90)
    longitude: float = Field(ge=-180, le=180)


class SavedLocationOut(BaseModel):
    id: uuid.UUID
    label: SavedLocationLabel
    custom_label: str | None
    address: str
    latitude: float
    longitude: float
    created_at: datetime

    model_config = {"from_attributes": True}


class LocationSearchResult(BaseModel):
    label: str
    subtitle: str | None
    latitude: float
    longitude: float
    kind: str  # ROAD | AREA | LANDMARK | CITY | HAZARD

File: citizen_app/apps/citizen-mobile/types/location.ts
export type SavedLocationLabel = 'HOME' | 'WORK' | 'COLLEGE' | 'CUSTOM';

export interface SavedLocation {
  id: string;
  label: SavedLocationLabel;
  customLabel: string | null;
  address: string;
  latitude: number;
  longitude: number;
  createdAt: string;
}

export interface CreateSavedLocationPayload {
  label: SavedLocationLabel;
  customLabel?: string;
  address: string;
  latitude: number;
  longitude: number;
}

export interface LocationSearchResult {
  label: string;
  subtitle: string | null;
  latitude: number;
  longitude: number;
  kind: 'ROAD' | 'AREA' | 'LANDMARK' | 'CITY' | 'HAZARD';
}
